pkg/reconciler/shared/tektonconfig/tektonconfig.go

```diff
@@ -239,6 +239,17 @@ func (r *Reconciler) markUpgrade(ctx context.Context, tc *v1alpha1.TektonConfig)
 		ChainProperties: chain,
 	}
 
+	// Deleting the `chains-config` configMap to handle the scenario when user upgrades
+	// i.e. in previous version `chains-config` configMap installerset was not there
+	// and with latest version we create separate installerset for `chains-config` configMap
+	// but the chains installerset will be deleted which initially had `chains-config` configMap
+	// and hence the configMap gets deleted and the deployment fails to get ready
+	if len(cm.Data) > 0 {
```

Comment:
this should not be handled here.

config reconciler will be trying to delete it and cains reconciler trying to create it, this should be handled in chains reconciler with better case handling

also cm.Data >0 is not the case we should be doing something specific